src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -318,6 +324,79 @@ def fuzz(self,
     return LibFuzzerCommon.fuzz(self, corpus_directories, fuzz_timeout,
                                 artifact_prefix, additional_args)
 
+class FuchsiaQemuLibFuzzerRunner(new_process.ProcessRunner,LibFuzzerCommon):
+  """libFuzzer runner (when Fuchsia is the target platform)."""
+  def __init__(self, executable_path, default_args=None):
+    # This code assumes the following layout for files in the storage bucket:
+    # * multiboot.bin
+    # * fvm.blk
+    # * fuchsia-ssh.zbi
+    # * qemu/*
+    # * .ssh/*
+    local_path = os.getcwd() + "/"
+    resources_path = local_path + "fuchsia_on_clusterfuzz_resources_v1"
```

Comment:
Downloading things in `libfuzzer.py` is a bit weird, I will need to think about where else this could go.

Reply:
yeah, this is a more high-level conversation i was interested in having with y'all :)

in the initial concept, i think we had the idea of downloading + launching QEMU being handled by launcher.py, but when i got into the details of that, my two main questions were:

* how should IPC be handled?  if i launch QEMU in launcher.py and run SSH in libfuzzer.py, then i need a way to communicate "use this portnumber and this directory location" to libfuzzer.py, and neither environment variables nor "hardcode those in as global constants" felt entirely right to me, and

* i think i must be confused about some of the functionality of launcher.py itself, because though I see "App launch command is [path_to_launcher.py] ." in my log messages, adding log lines to launcher.py:main() seemed to have no effect, which means either I was goofing where I was looking for logs, OR, launcher.py isn't being called the way i expect

at that point I realized it may be more efficient to throw something up, and then see where you all think the best place for such matters is :)

Reply:
>and neither environment variables nor "hardcode those in as global constants" felt entirely right to me, and

We do use environment variables for this sort of thing.

>" in my log messages, adding log lines to launcher.py:main() seemed to have no effect,

I think these logs show up in run_fuzzer.log rather than bot.log. launcher.py gets called in a new process.

>at that point I realized it may be more efficient to throw something up, and then see where you all think the best place for such matters is :)

Not a bad idea. I'm still thinking about whether this should land as is and be fixed later or fixed first.